### PR TITLE
Zapier plugin

### PR DESCRIPTION
**Pull request by escopecz**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | https://mautic.org/docs/en/plugins/zapier.html
| Related developer documentation PR URL | https://github.com/mautic/mautic-zapier, https://github.com/mautic/developer-documentation/pull/98
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This plugin adds very simple Zapier plugin which will just shows a list of preconfigured Zap templates, adds a link to docs and mention the API must be enabled.

To do this easily I added "Secondary Plugin Description". Until now the description text showed up only in the H3 showing in 8/12 of the popup. It wasn't convenient to show the Zap templates there. So I implemented simple solution. If the description is separated by `\n---\n` (`\n` means new line) then the part after this separator will show up below the logo and H3 in the full width. The separator was taken from the YAML syntax where it "indicates the start of a document".

#### Steps to test this PR:
1. Go to plugins.
2. Hit install.
3. The Zapier plugin will show up.
4. Open it.
5. This is what you should see:

screen shot 2018-03-09 at 12 10 43

It will have more than one Zap template in the near future.

### CODE REVIEW

**Comment by kuzmany** [comment]
Tested by https://github.com/mautic/mautic-tester
Works 👍 

UI is little different like in your example:
image

**Comment by Maxell92** [comment]
Works for me, but I am not sure if this test is enough? Is this really all I should test? Only that this screen will be shown?

**Comment by kuzmany** [comment]
@Maxell92 Yes, plugin is just shortcut to Zapier templates on external site

**Comment by escopecz** [comment]
The description must look as my picture in the description. The first line as H3 and the rest as normal test bellow. I'll try to uninstall and install again if I'll be able to replicate @kuzmany's issue.

**Comment by escopecz** [comment]
Nope, it works for me even if I delete the plugin and install it again. This is the result:
screen shot 2018-03-19 at 11 38 38

As mentioned in the description the `\n---\n` should divide the description into header and the rest. @kuzmany could you check if the Zapier plugin description in the plugins table has the new lines before and after `---`? It does for me. If that won't be a stable solution then I'll remove the new line requirement.

@Maxell92 can you confirm if the description is divided for you?

**Comment by Maxell92** [comment]
Works for me.

screen shot 2018-03-19 at 13 04 04

File: app/bundles/PluginBundle/Entity/Plugin.php [hunk 1]
```diff
@@ -21,6 +21,8 @@
  */
 class Plugin extends CommonEntity
 {
+    const DESCRIPTION_DELIMITER = "\n---\n";
```
Comment (kuzmany) [inline app/bundles/PluginBundle/Entity/Plugin.php]: For windows support just add, then will works properly

` const DESCRIPTION_DELIMITER = "\r\n---\r\n";`

image

**Comment by escopecz** [comment]
@kuzmany please retest. I also added a unit test with Windows line endings.

**Comment by javjim-mautic** [comment]
Works for me

**Pull request merged**